inc/class-book.php

```diff
@@ -109,7 +109,16 @@ static function getBookInformation( $id = null, $contributors_as_string = true,
 		if ( static::useCache() ) {
 			$cached_book_information = wp_cache_get( $cache_id, 'pb' );
 			if ( $cached_book_information ) {
-				if ( ! $read_contributors_from_cache ) {
+				$contributors_cached_as_string = true;
+				foreach ( $contributors->valid as $contributor_type ) {
+					if (
+						isset( $cached_book_information[ $contributor_type ] )
+					) {
+						$contributors_cached_as_string = is_string( $cached_book_information[ $contributor_type ] );
+						break;
+					}
+				}
```

Comment:
As contributor types are dynamic and sometimes some of those those could be absents in different books, we need to check in all types. It will return at the first existing contributor type stored in the cache.